### PR TITLE
[agw][lte][bug fix] Fix for issue 6325

### PR DESCRIPTION
**Pull request by pruthvihebbani**
[agw][lte][bug fix] Fix for issue 6325

## Summary

This PR fixes one of the issues in #6325 where mme was accepting attach complete message when it was in the process of implicit detach due to ICS timer expiry.

## Test Plan

- Added a new TC to simulate sending of attach complete message after ICS timer expires
- Verified s1ap tester sanity

### CODE REVIEW

File: lte/gateway/c/oai/tasks/nas/emm/Attach.c [hunk 1]
```diff
@@ -710,6 +710,15 @@ int emm_proc_attach_complete(
           (nas_emm_attach_proc_t*)
               ue_mm_context->emm_context.emm_procedures->emm_specific_proc;
 
+      /* Process attach complete msg only if T3450 timer is running
+       * If its not running it means that implicit detach is in progress
+       */
+      if (attach_proc->T3450.id == NAS_TIMER_INACTIVE_ID) {
+        OAILOG_WARNING(
```
Comment (ardzoht) [inline lte/gateway/c/oai/tasks/nas/emm/Attach.c]: nit: can we use `OAILOG_WARNING_UE` to prepend the IMSI here? would help for tracing certain subscribers

Comment (ulaskozat) [reply lte/gateway/c/oai/tasks/nas/emm/Attach.c]: second that...

**Review by ulaskozat — APPROVED** [review]
LGTM. Please address the nit comment.

**Review by ardzoht — APPROVED** [review]

File: lte/gateway/c/oai/tasks/nas/emm/Attach.c [hunk 2]
```diff
@@ -710,6 +710,18 @@ int emm_proc_attach_complete(
...
+      /* Process attach complete msg only if T3450 timer is running
+       * If its not running it means that implicit detach is in progress
+       */
+      if (attach_proc->T3450.id == NAS_TIMER_INACTIVE_ID) {
+        OAILOG_WARNING_UE(
+            LOG_NAS_EMM, ue_mm_context->emm_context._imsi64,
+            "Discarding attach complete as T3450 timer is not active for "
+            "ueid " MME_UE_S1AP_ID_FMT "\n",
+            ue_id);
+        bdestroy((bstring)(esm_msg_pP));
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/nas/emm/Attach.c]: isn't this a violation of constness of this parameter? Shall we instead return error and let caller destroy it?

Comment (pruthvihebbani) [reply lte/gateway/c/oai/tasks/nas/emm/Attach.c]: fixed

**Pull request merged**